### PR TITLE
feat(header|syncer)!: add params to syncer

### PR DESCRIPTION
**Pull request by vgonkivs**
## Overview
Resolves https://github.com/celestiaorg/celestia-node/issues/1543

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

### CODE REVIEW

**Comment by Wondertan** [comment]
Unit test for syncer broke

File: header/sync/options.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import (
+	"fmt"
+	"time"
+)
+
+type Options func(*Parameters)
+
+// Parameters is the set of parameters that must be configured for the syncer.
+type Parameters struct {
+	// blockTime provides a reference point for the Syncer to determine
```
Comment (distractedm1nd) [inline header/sync/options.go:12]:
```suggestion
	// BlockTime provides a reference point for the Syncer to determine
```

**Review by distractedm1nd — CHANGES_REQUESTED** [review]

File: header/sync/sync.go [hunk 2]
```diff
@@ -54,17 +50,27 @@ type Syncer struct {
 	// controls lifecycle for syncLoop
 	ctx    context.Context
 	cancel context.CancelFunc
+
+	Params *Parameters
```
[the remark this answers is not in the document]
Comment (Wondertan) [reply header/sync/sync.go:54]: So we can unit tests that they are correctly applied in module tests. We have the same for Store tests.

**Review by distractedm1nd — APPROVED** [review]

**Review by renaynay — APPROVED** [review]

File: header/sync/options.go [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+	"time"
+)
+
+type Options func(*Parameters)
+
+// Parameters is the set of parameters that must be configured for the syncer.
+type Parameters struct {
+	// BlockTime provides a reference point for the Syncer to determine
+	// whether its subjective head is outdated
+	BlockTime time.Duration
```
Comment (Wondertan) [inline header/sync/options.go:14]: BlockTime is a global network variable and something other than something users should configure independently. Let's keep it private and set it in the nodebuilder from p2p module where it's currently defined

File: header/sync/options.go [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+		MaxRequestSize: 512,
+	}
+}
+
+func (p *Parameters) Validate() error {
+	if p.BlockTime == 0 {
+		return fmt.Errorf("invalid block time duration: %v", p.BlockTime)
+	}
+	if p.TrustingPeriod == 0 {
+		return fmt.Errorf("invalid trusted time duration: %v", p.TrustingPeriod)
```
Comment (Wondertan) [inline header/sync/options.go:42]:
```suggestion
		return fmt.Errorf("invalid trusting period duration: %v", p.TrustingPeriod)
```

File: header/sync/options.go (hunk 3 above)
Comment (Wondertan) [reply header/sync/options.go:14]: I see you already set it from modp2p. Let's just privatize this then

**Review by Wondertan — DISMISSED** [review]

**Review by Wondertan — DISMISSED** [review]

**Review by renaynay — APPROVED** [review]
optional nit (but recommended) is to move `blockTime` to bottom of params struct and document why it's private and why a 0 value is okay.

**Review by renaynay — APPROVED** [review]

**Review by Wondertan — APPROVED** [review]

**Pull request merged**